### PR TITLE
updated Downloader with ifUpdateAvailable

### PR DESCRIPTION
**Pull request by plata**
requires PhoenicisOrg/POL-POM-5#979

If the download location is specified (`to()`), the new (`ifUpdateAvailable()`) method can be used to trigger a download only if the resource available via the given URL is newer than the local file.

It is currently used to avoid repeated downloads of the available Wine versions json.

### CODE REVIEW

**Comment by plata** [comment]
@qparis the last modified date reported for <http://phoenicis.playonlinux.com/index.php/wine?os=linux> is always 0. Can you change the server to return the correct date?

File: Utils/Functions/Net/Download/script.js [hunk 1]
```diff
@@ -32,6 +32,10 @@ var Downloader = function () {
         that._localFile = localFile;
         return that;
     };
+    that.ifUpdateAvailable = function (ifUpdateAvailable) {
```
Comment (madoar) [inline Utils/Functions/Net/Download/script.js]: What exactly is the use of this function/field `ifUpdateAvailable`?

Comment (plata) [inline Utils/Functions/Net/Download/script.js]: Like I wrote in the description: It allows you to only download the file again if the file on the web server is newer than the one you have downloaded (i.e. it has been updated on the server).

Comment (madoar) [inline Utils/Functions/Net/Download/script.js]: I'm a little confused about the type of the variable `ifUpdateAvailable` that should be passed to the function.
When taking a look at normal java APIs, like https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html I normally think that methods beginning with an `if` take a lambda function,  but I believe your method takes a boolean right?

Comment (plata) [inline Utils/Functions/Net/Download/script.js]: Yes, it takes a boolean. Any proposal how it should be named?

Comment (madoar) [inline Utils/Functions/Net/Download/script.js]: Maybe something like `alwaysCheckForUpdates` or only `checkForUpdates`?

**Pull request merged**